src/foam/nanos/notification/email/EmailTemplateEngine.js

```diff
@@ -305,28 +294,33 @@ foam.CLASS({
     {
       name: 'renderTemplateById',
       args: [
-        { name: 'id', type: 'String' }
+        { name: 'id', type: 'String' },
+        { name: 'values', type: 'Map' },
+        { name: 'x', type: 'Context' }
       ],
       type: 'StringBuilder',
       javaCode: `
       EmailTemplate template = (EmailTemplate) ((DAO) x_.get("emailTemplateDAO")).find(id);
-      return renderTemplate(template.getBody());
+      return renderTemplate(template.getBody(), values, x);
       `
     },
     {
       name: 'renderTemplate',
       args: [
-        { name: 'str', type: 'String' }
+        { name: 'str', type: 'String' },
+        { name: 'values', type: 'Map' },
+        { name: 'x', type: 'Context' }
```

Comment:
Can you make this the first arg?

Reply:
I did. Please reload the page

Reply:
you mean, make context the first argument?